### ISSUE TITLE
scrapy different with requests? requests work, but scrapy not

### ISSUE THREAD

**Issue by zeroleo12345** [scrapy/scrapy#3388]
1. use requests library, it work. 
``` 
import requests

headers = {}
headers['Host'] = 'www.bloomberg.com'
headers['User-Agent'] = 'Charles/4.2.1'

response = requests.get('https://www.xxx.com/quote/700:HK', headers=headers
```

2. When use scrapy, it was block because the Website realize scrapy is a robot. what's the different with requests?

2018-08-15 16:48:15 [scrapy.downloadermiddlewares.redirect] DEBUG: Redirecting (307) to <GET https://www.xxx.com/tosv2.html?vid=&uuid=f3262880-a067-11e8-922c-0f4bf8a1cda2&url=L3F1b3RlLzcwMDpISw==> from <GET https://www.xxx.com/quote/700:HK>

**Comment by wushaojun321** [comment]
do you modify ROBOTSTXT_OBEY = False in setting.py?

**Comment by zeroleo12345** [comment]
yes，i modify it。and i just crawl one page，it occur the log above

**Comment by wRAR** [comment]
Hello @zeroleo12345,thank you for forgetting to remove the website name from the code, it allowed me to play with this for some time. It's definitely an anti-bot protection but I don't know why it trips on Scrapy requests more often than on similar curl requests. I say "more often" because I got both 200 with Scrapy and 307 with curl. You should have more luck by setting a real user-agent but it's definitely not enough. My only guess is that Scrapy does something with HTTPS differently because it seems the only thing that can be different when all headers are the same.

**Comment by danielnegrini** [comment]
I have the some problem here. @zeroleo12345 Did you find out what the difference is?

**Comment by allanmachado1** [comment]
My guess is this problem is related to #1854

**Comment by EdgarMagalhaes** [comment]
I get the same problem when trying to crawl HTTPS. Any developments so far?

**Comment by wRAR** [comment]
@EdgarMagalhaes is it for bloomberg or for some other website? The bloomberg case is special as it's most likely a sophisticated anti-bot measure.

**Comment by EdgarMagalhaes** [comment]
Its for _https://mobile.de_
I can post the "not-working" code if you'd like... thank you for replying so quickly.

**Comment by wRAR** [comment]
@EdgarMagalhaes plain `scrapy shell https://www.mobile.de` works for me so it's likely some local problem, old OpenSSL or broken configuration.

**Comment by EdgarMagalhaes** [comment]
@wRAR Thank you again.
Ok, so let me delve into the problem so you can get more information.

Mobile.de is a used car buying/selling platform. I'm trying to scrape a specific search on that site, like this one: [Mobile.de - Search](https://suchen.mobile.de/fahrzeuge/search.html?dam=0&isSearchRequest=true&ms=25100&sfmr=false&vc=Car)

If you try to scrape it directly it will answer with "Are you a human?" page.

So we need to trick it, it needs specific cookies, 7 to be exact. The steps to get all the cookies are:
1 - GET Request to [](https://mobile.de) witch returns 6 cookies
2 - GET Request to [](https://suchen.mobile.de/ ) witch returns the 7th cookie
3 - GET Request to [](https://suchen.mobile.de/fahrzeuge/search.html?dam=0&isSearchRequest=true&ms=25100&sfmr=false&vc=Car)

Running Step 3 with Scrapy returns the "Are you a human?" page. Running with Requests module returns the page with the actual data to be scraped.

**Comment by wRAR** [comment]
@EdgarMagalhaes as it involves a captcha it may be a sophisticated anti-bot measure too, and it can, for example, require a specific order of headers or other things that should not matter usually. There is a lot of such things that different HTTP clients can do differently.

**Comment by Gallaecio** [comment]
Closing as this is a problem not with Scrapy itself.

Please, use [StackOverflow](https://stackoverflow.com/tags/scrapy) to ask this type of questions. See [Getting Help](https://docs.scrapy.org/en/latest/#getting-help).

Also, https://github.com/scrapy/scrapy/pull/3862 (to be included in the upcoming Scrapy 1.8.0) may come in handy for these situations.
